### ISSUE TITLE
[Feature Request] Add support for validation dataset during training

### ISSUE THREAD

**Issue by piyifan123** [ByteDance-Seed/VeOmni#247]
#### Description
Add support for validation dataset including

-  dataloader builder for validation dataset
-  subloop that can be plugged in trainers to run validation dataset to compute cross entropy loss
    -  note that the generation stage is not needed as we only care about the standard cross entropy loss instead of a reward.
-  log validation loss in wandb.
-  add options and plug in the validation subloop to trainers.
-  a sample task to run sft with validation and show validation loss decreases as the training goes on.

#### Reference

An internal implementation was there and we can reference from it. [Link](https://code.byted.org/data/VeOmni/blob/master/tasks/text/train_torch.py?ref_type=heads) Note: only Bytedancers can open it.

But we want to standardize the impl as a package under veomni/ package so that it's cleaner for trainers to plug in and we can add unit tests.

### Motivation

The current implementation lacks a mechanism to evaluate the model on a hold-out set during training. I would like to request support for a validation dataset to enable real-time monitoring of generalization performance and to facilitate valid model selection/checkpointing strategies.

This will also be convenient to perform ablation studies via veOmni.

### Your contribution

N/A (created as a batch of planned features, will check the community for contributors later)
